openeo/local/processing.py

```diff
@@ -6,22 +6,36 @@
 from pathlib import Path
 
 from openeo_pg_parser_networkx import ProcessRegistry
-from openeo_processes_dask.core import process as openeo_process
+from openeo_processes_dask.process_implementations.core import process
+import openeo_processes_dask.specs
+import openeo_processes_dask.process_implementations
+from openeo_pg_parser_networkx.process_registry import Process
 
-standard_processes = [
-    func
-    for _, func in inspect.getmembers(
-        importlib.import_module("openeo_processes_dask.process_implementations"),
-        inspect.isfunction,
-    )
-]
+def init_process_registry():
+    process_registry = ProcessRegistry(wrap_funcs=[process])
 
-PROCESS_REGISTRY = ProcessRegistry(wrap_funcs=[openeo_process])
+    # Import these pre-defined processes from openeo_processes_dask and register them into registry
+    processes_from_module = [
+        func
+        for _, func in inspect.getmembers(
+            openeo_processes_dask.process_implementations,
+            inspect.isfunction,
+        )
+    ]
 
-for func in standard_processes:
-    PROCESS_REGISTRY[func.__name__] = func
+    specs = {
+        func.__name__: getattr(openeo_processes_dask.specs, func.__name__)
+        for func in processes_from_module
+    }
+
+    for func in processes_from_module:
+        process_registry[func.__name__] = Process(
+            spec=specs[func.__name__], implementation=func
+        )
+    return process_registry
```

Comment:
Unless I'm missing something, I think this can be done more compactly with:

```python
for name, func in inspect.getmembers(
    openeo_processes_dask.process_implementations,
    inspect.isfunction,
):
    process_registry[name] = Process(
        spec=getattr(openeo_processes.specs, name),
        implementation=func
    )
```

Also, wouldn't it be better that openeo_processes_dask provide helpers to list implementations/spec, instead of having to use ugly `inspect.getmembers` and `getattr` constructs?

Reply:
Ah apologies, I missed this tag - yes, agree with Stefaan, this is something openeo-processes-dask could do better - have created an issue! 
https://github.com/Open-EO/openeo-processes-dask/issues/92